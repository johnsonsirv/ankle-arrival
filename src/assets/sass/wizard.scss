.wizardPanel {
    display: flex;
    flex-direction: column;
    align-items: center;
    & button {
        width: 12vw;
        height: 6vh;
        cursor: pointer;
    }
}
.wizardSteps {
    display: flex;
    flex-direction: column;
    align-items: center;
    font-size: 13px;
    border: 1px solid #d8d4d4;
    margin: 9px;
    text-align: left;
    padding: 9px;
    line-height: 1.5em;
    letter-spacing: .1em;
    border-radius: 5px;
}

.injuryPanel {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    width: 520px;
    height: 160px;
    margin-top: 10%;
    padding: 9px;
    border: 1px solid #d8d4d4;
    border-radius: 5px;
    & select {
        width: 300px;
        height: 50px;
        outline: none;
        border-style: none;
        border-radius: 3px;
        font-size: 22px;
        padding: 4px;
        text-transform: capitalize;
        & option {
            outline: none;
        }
    }
}

.symptomsPanel {
    display: flex;
    flex-direction: column;
    text-align: left;
    justify-content: space-around;
    height: 450px;
    margin-top: 4%;
    padding: 9px;
    border: 1px solid #d8d4d4;
    border-radius: 5px;
    & p {
        text-align: center;
    }
    & label {
        display: flex;
        font-size: 13px;
        text-transform: capitalize;
        align-items: center;
        height: 50px;
        background-color: #f5ededf1;

        & input[type="checkbox"] {
            width: 28px;
            height: 28px;
            outline-style: solid;
            outline-color: #817979;
            outline-width: .2em;
            margin-right: 6px;
            
            &:checked {
                outline-style: unset;
            }
        }   
        &:hover {
            cursor: pointer;
        }
    }
    & .injuryActions {
        display: flex;
        justify-content: space-evenly;
    }

}

.biodataPanel {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    width: 520px;
    height: 180px;
    margin-top: 10%;
    padding: 9px;
    border: 1px solid #d8d4d4;
    border-radius: 5px;
    & p {
        text-align: center;
    }

    & input[type=text] {
        width: 300px;
        height: 50px;
        border-style: solid;
        border-color: #d8d4d4;
        border-radius: 3px;
        font-size: 22px;
        padding: 5px;
        text-transform: capitalize;
    }

    & .biodataActions {
        display: flex;
        width: 100%;
        justify-content: space-evenly;
    }
}

.diagnosisPanel {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    margin-top: 3%;
    padding: 9px;
    border: 1px solid #d8d4d4;
    text-transform: capitalize;
}