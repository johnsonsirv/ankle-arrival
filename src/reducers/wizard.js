import {
  WIZARD_RECEIVE_INJURIES,
  WIZARD_REQUEST_INJURIES,
  WIZARD_REQUEST_SYMPTOMS,
  WIZARD_RECEIVE_SYMPTOMS,
  WIZARD_REQUEST_DIAGNOSIS,
  WIZARD_RECEIVE_DIAGNOSIS,
  WIZARD_NEXT_STEP,
} from '../actions/actionTypes';

const initialState = {
  isFetching: false,
};

const wizard = (state = initialState, action) => {
  switch (action.type) {
    case WIZARD_REQUEST_INJURIES:
      return {
        ...state,
        isFetching: true,
      };
    case WIZARD_RECEIVE_INJURIES:
      return {
        ...state,
        injuries: [...action.payload],
        isFetching: false,
      };
    case WIZARD_REQUEST_SYMPTOMS:
      return {
        ...state,
        isFetching: true,
      };
    case WIZARD_RECEIVE_SYMPTOMS:
      return {
        ...state,
        symptoms: [...action.payload],
        isFetching: false,
      };
    case WIZARD_REQUEST_DIAGNOSIS:
      return {
        ...state,
        isFetching: true,
      };
    case WIZARD_RECEIVE_DIAGNOSIS:
      return {
        ...state,
        diagnosis: { ...action.payload },
        isFetching: false,
      };
    case WIZARD_NEXT_STEP:
      return {
        ...state,
        next: { ...action.payload },
      };
    default:
      return state;
  }
};

export default wizard;
