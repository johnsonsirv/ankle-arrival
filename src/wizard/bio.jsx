import React from 'react';
import PropTypes from 'prop-types';
import NumberTextField from '../components/forms/inputs/numberTextField';
import InputTextField from '../components/forms/inputs/inputTextField';
import ComboBox from '../components/forms/inputs/comboBox';

const GENDER = [
  { id: 1, value: '', text: '' },
  { id: 2, value: 'female', text: 'Female' },
  { id: 3, value: 'male', text: 'Male' },
];

const BioPage = ({ userBio, onInputChange }) => (
  <div>
    <form>
      <InputTextField
        id="fullname"
        name="fullname"
        onChange={onInputChange}
        value={userBio.fullname}
      />
      <NumberTextField
        id="age"
        name="age"
        onChange={onInputChange}
        value={userBio.age}
      />
      <InputTextField
        id="height"
        name="height"
        onChange={onInputChange}
        value={userBio.height}
      />
      <ComboBox onChange={onInputChange}>
        {GENDER.map(item => (
          <option key={item.id} value={item.value}>
            {item.text}
          </option>
        ))}
      </ComboBox>
    </form>
  </div>
);

BioPage.propTypes = {
  userBio: PropTypes.shape({
    fullname: PropTypes.string,
    age: PropTypes.string,
    height: PropTypes.string,
  }).isRequired,
  onInputChange: PropTypes.func.isRequired,
};

export default BioPage;
