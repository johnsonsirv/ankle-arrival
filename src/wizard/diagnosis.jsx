import React from 'react';
import PropTypes from 'prop-types';

const DiagnosisPage = ({
  diagnosis: {
    injury,
    disease,
    symptoms,
    player,
    inference,
    treatment,
    lifestyle,
  },
}) => (
  <div>
    <header>Results</header>
    <header>Diagnosis &amp; symptoms</header>
    <div>
      <p>{injury}</p>
    </div>
    <div>
      <p>{symptoms}</p>
    </div>
    <header>Possible diseases based on infered analysis</header>
    <div>
      <p>{disease}</p>
    </div>
    <header>Treatment / Lifestyle</header>
    <div>
      <p>{treatment}</p>
    </div>
    <div>
      <p>{lifestyle}</p>
    </div>
  </div>
);

DiagnosisPage.propTypes = {
  diagnosis: PropTypes.shape({
    id: PropTypes.number,
    injury: PropTypes.string,
    disease: PropTypes.string,
    symptoms: PropTypes.string,
    player: PropTypes.string,
    inference: PropTypes.string,
    treatment: PropTypes.string,
    lifestyle: PropTypes.string,
  }).isRequired,
};

export default DiagnosisPage;
