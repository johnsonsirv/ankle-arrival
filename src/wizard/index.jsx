import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Spinner from 'react-spinkit';
import * as dispatchActions from '../actions';
import WizardSteps from './wizardSteps';
import InjuryPage from './injury';
import SymptomsPage from './symptoms';
import BioPage from './bio';
import Button from '../components/forms/inputs/button';
import DiagnosisPage from './diagnosis';

const mapStateToProps = state => state;

export const Wizard = props => {
  const [wizard, setWizard] = useState({
    showUserBioPage: false,
    init: true,
    selectedInjury: null,
    userBio: {
      fullname: '',
      age: '',
      height: '',
    },
  });

  const handleStartDiagnosis = () => {
    props.fetchInjuries();
    wizard.init = false;
    setWizard(wizard);
  };

  const handleShowSymptoms = () => {
    props.fetchSymptoms();
  };

  const handleShowBioPage = () => {
    wizard.showUserBioPage = true;
    setWizard(wizard);
  };

  const handleSubmitDiagnosis = () => {
    props.fetchDiagnosis();
    wizard.showUserBioPage = true;
    setWizard(wizard);
  };

  const handleRestartDiagnosis = () => {
    wizard.init = true;
    setWizard(wizard);
  };

  const handleChange = () => {};

  const handleClickSymptoms = () => {};

  const { isFetching, injuries, symptoms, diagnosis } = props;
  return (
    <>
      {isFetching && <Spinner name="three-bounce" fadeIn="none" />}
      {wizard.init && (
        <>
          <WizardSteps />
          <Button
            onClick={handleStartDiagnosis}
            value="Start Diagnosis"
            disabled={!wizard.init}
            id="step-1-start-diagnosis"
          />
        </>
      )}
      {injuries && (
        <>
          <InjuryPage injuries={injuries} onChange={handleChange} />
          <Button
            onClick={handleShowSymptoms}
            value="Next"
            disabled={!injuries}
            id="step-2-symptoms"
          />
        </>
      )}
      {symptoms && (
        <>
          <SymptomsPage symptoms={symptoms} onClick={handleClickSymptoms} />
          <Button
            onClick={handleShowBioPage}
            value="Next"
            disabled={!symptoms}
            id="step-3-bio"
          />
        </>
      )}
      {wizard.showUserBioPage && (
        <>
          <BioPage userBio={wizard.userBio} onInputChange={handleChange} />
          <Button
            onClick={handleSubmitDiagnosis}
            value="Submit"
            disabled={!wizard.showUserBioPage}
            id="step-4-submit"
          />
        </>
      )}
      {diagnosis && (
        <>
          <DiagnosisPage diagnosis={diagnosis} />
          <Button
            onClick={handleRestartDiagnosis}
            value="Re-start Diagnosis"
            id="step-5-restart-diagnosis"
          />
        </>
      )}
    </>
  );
};

Wizard.propTypes = {
  fetchSymptoms: PropTypes.func.isRequired,
  fetchInjuries: PropTypes.func.isRequired,
  fetchDiagnosis: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  injuries: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      code: PropTypes.number,
      name: PropTypes.string,
    }).isRequired
  ).isRequired,
  symptoms: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      code: PropTypes.string,
      description: PropTypes.string,
    }).isRequired
  ).isRequired,
  diagnosis: PropTypes.shape({
    id: PropTypes.number,
    injury: PropTypes.string,
    disease: PropTypes.string,
    symptoms: PropTypes.string,
    player: PropTypes.string,
    inference: PropTypes.string,
    treatment: PropTypes.string,
    lifestyle: PropTypes.string,
  }).isRequired,
};

export default connect(mapStateToProps, dispatchActions)(Wizard);
