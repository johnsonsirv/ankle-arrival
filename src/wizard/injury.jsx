import React from 'react';
import PropTypes from 'prop-types';
import ComboBox from '../components/forms/inputs/comboBox';

const InjuryPage = ({ injuries, onChange }) => (
  <ComboBox onChange={onChange}>
    {injuries.map(item => (
      <option key={item.id} value={item.code}>
        {item.name}
      </option>
    ))}
  </ComboBox>
);

InjuryPage.propTypes = {
  injuries: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      code: PropTypes.number,
      name: PropTypes.string,
    }).isRequired
  ).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default InjuryPage;
