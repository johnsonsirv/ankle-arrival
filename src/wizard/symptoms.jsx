import React from 'react';
import PropTypes from 'prop-types';
import ComboBox from '../components/forms/inputs/comboBox';

const SymptomsPage = ({ injuries, onChange }) => (
  <ComboBox onChange={onChange}>
    {injuries.map(item => (
      <option key={item.id} value={item.code}>
        {item.name}
      </option>
    ))}
  </ComboBox>
);

SymptomsPage.propTypes = {
  injuries: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      code: PropTypes.string,
      description: PropTypes.string,
    }).isRequired
  ).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default SymptomsPage;
