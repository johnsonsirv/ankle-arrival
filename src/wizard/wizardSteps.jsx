import React from 'react';

const WizardSteps = () => {
  return (
    <div>
      <div>Step 1</div>
      <div>Step 2</div>
      <div>Step 3</div>
    </div>
  );
};

export default WizardSteps;
