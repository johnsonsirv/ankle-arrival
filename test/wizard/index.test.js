/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
// import { BrowserRouter as Router, Redirect } from 'react-router-dom';
import { Wizard } from '../../src/wizard/index';

function setup(stateProps = {}) {
  const props = {
    fetchInjuries: jest.fn(),
    fetchSymptoms: jest.fn(),
    fetchDiagnosis: jest.fn(),
    isFetching: false,
    injuries: [],
    symptoms: [],
    diagnosis: {},
  };

  const enzymeWrapper = shallow(<Wizard {...props} {...stateProps} />);

  return { props, enzymeWrapper };
}

describe('<Wizard /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('should render correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 <Button> to start diagnosis', () => {
    expect(enzymeWrapper.find('#step-1-start-diagnosis')).toHaveLength(1);
  });
  it('should render <WizardSteps />', () => {
    expect(enzymeWrapper.find('WizardSteps')).toHaveLength(1);
  });
  it('should render <InjuryPage> when state changes', () => {
    const injuries = [
      { id: 1, name: 'abdominal', code: 1 },
      { id: 2, name: 'bone', code: 2 },
    ];
    const { enzymeWrapper } = setup({ injuries });
    expect(enzymeWrapper.find('InjuryPage')).toHaveLength(1);
  });
  it('should render <SymptomsPage> when state changes', () => {
    const symptoms = [
      { id: 5, code: 'a14', description: 'excessive abdominal' },
    ];
    const { enzymeWrapper } = setup({ symptoms });
    expect(enzymeWrapper.find('SymptomsPage')).toHaveLength(1);
  });
  it('should render <BioPage> when state changes', () => {
    const symptoms = [
      { id: 5, code: 'a14', description: 'excessive abdominal' },
    ];
    const { enzymeWrapper } = setup({ symptoms });
    enzymeWrapper
      .find('#step-3-bio')
      .simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(enzymeWrapper.find('BioPage')).toHaveLength(1);
  });
  it('should render <DiagnosisPage> when state changes', () => {
    const diagnosis = {
      id: 18,
      injury: 'abdominal',
      disease: 'blunt abdominal injury',
      symptoms: 'excessive abdominal',
      player: 'Joe Smith',
      inference:
        'if player exhibits symptoms excessive abdominal, in abdominal affected area, then there is a blunt abdominal injury disease.',
      treatment: 'surgery',
      lifestyle: 'maintain healthy lifestyle',
    };
    const { enzymeWrapper } = setup({ diagnosis });
    expect(enzymeWrapper.find('DiagnosisPage')).toHaveLength(1);
  });
});

describe('<Wizard /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('STEP 1 <start-diagnosis> should dispatch fetchInjuries action', () => {
    enzymeWrapper
      .find('#step-1-start-diagnosis')
      .simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(props.fetchInjuries).toHaveBeenCalled();
  });
  it('STEP 2 <symptoms> should dispatch fetchSymptoms action', () => {
    const injuries = [
      { id: 1, name: 'abdominal', code: 1 },
      { id: 2, name: 'bone', code: 2 },
    ];
    const { props, enzymeWrapper } = setup({ injuries });
    enzymeWrapper
      .find('#step-2-symptoms')
      .simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(props.fetchSymptoms).toHaveBeenCalled();
  });

  it('STEP 4 <submit> should dispatch fetchDiagnosis', () => {
    const symptoms = [
      { id: 5, code: 'a14', description: 'excessive abdominal' },
    ];
    const { props, enzymeWrapper } = setup({ symptoms });
    enzymeWrapper
      .find('#step-3-bio')
      .simulate('click', { target: {}, preventDefault: jest.fn() });
    enzymeWrapper
      .find('#step-4-submit')
      .simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(props.fetchDiagnosis).toHaveBeenCalled();
  });
});
