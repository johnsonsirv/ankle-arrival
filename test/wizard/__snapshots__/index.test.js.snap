// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wizard /> rendering should render correctly 1`] = `ShallowWrapper {}`;
